contracts/receivables/BaseCreditPoolReceivable.sol

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import {BaseStructs as BS} from "../libraries/BaseStructs.sol";
+import "../BaseCreditPool.sol";
+
+/**
+ * @title BaseCreditPoolReceivable
+ * @dev Contract that represents an ERC721 receivable payable to Huma's BaseCreditPool
+ */
+contract BaseCreditPoolReceivable is ERC721Enumerable, ERC721URIStorage, AccessControl {
+    using Counters for Counters.Counter;
+    using BS for BS.CreditRecord;
+
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    Counters.Counter private _tokenIdCounter;
+
+    struct ReceivableInfo {
+        // The BaseCreditPool that's paid by the receivable
+        BaseCreditPool baseCreditPool;
+        // The payment method used to settle the receivable
+        PaymentMethod paymentMethod;
+        // The ERC20 token used to settle the receivable
+        address paymentToken;
+        // The total expected payment amount of the receivable
+        uint96 receivableAmount;
+        // The amount of the receivable that has been paid so far
+        uint96 balance;
+        // The date at which the receivable becomes due
+        uint64 maturityDate;
+    }
+
+    // Map tokenId to receivable information
+    mapping(uint256 => ReceivableInfo) public receivableInfoMapping;
+
+    // The status of the receivable.
+    enum Status {
+        Unpaid,
+        Paid,
+        PartiallyPaid
+    }
+
+    // The payment method used to settle the receivable
+    enum PaymentMethod {
+        // Payment is declared by calling the declarePayment function on this contract.
+        // No ERC20 tokens are transferred in this call. The receivable owner is responsible
+        // for transferring the ERC20 tokens to the pool on their own.
+        Declarative,
+        // Payment is made by transferring ERC20 tokens to the pool
+        // using the makePayment function on the BaseCreditPool contract.
+        Payable
```

Comment:
Since we are not going to handle makePayment in this contract, this is just a placeholder. Recommend you comment it for future expansion. 

payable is a keyword for solidity.  Payable will be really confusing. Please consider a different name.

Reply:
I will just remove this, I don't like keeping comments around that "might" be used in the future.